diffusion/models/autoencoder.py

```diff
@@ -307,32 +356,54 @@ def __init__(self,
         self.post_quant_conv = nn.Conv2d(self.latent_channels, self.latent_channels, kernel_size=1, stride=1, padding=0)
         nn.init.kaiming_normal_(self.post_quant_conv.weight, nonlinearity='linear')
 
+    @property
+    def device(self) -> torch.device:
+        return next(self.parameters()).device
+
+    def get_extra_state(self):
+        return {'config': self.config}
+
+    def set_extra_state(self, state):
+        # Save the autoencoder config
+        config = {}
+        config['input_channels'] = self.input_channels
+        config['output_channels'] = self.output_channels
+        config['hidden_channels'] = self.hidden_channels
+        config['latent_channels'] = self.latent_channels
+        config['double_latent_channels'] = self.double_latent_channels
+        config['channel_multipliers'] = self.channel_multipliers
+        config['num_residual_blocks'] = self.num_residual_blocks
+        config['use_conv_shortcut'] = self.use_conv_shortcut
+        config['dropout_probability'] = self.dropout_probability
+        config['resample_with_conv'] = self.resample_with_conv
+        config['use_attention'] = self.use_attention
+        config['zero_init_last'] = self.zero_init_last
```

Comment:
Why not just construct this config sttically in one go?

Reply:
Agreed. Fixed it.